### PR TITLE
Programmes 6706 initial clip list page

### PR DESCRIPTION
**Pull request by cisco87**
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6707

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: config/routes/3rd_party.yaml [hunk 1]
```diff
@@ -77,11 +77,11 @@ categories_show:
     requirements: { categoryType: 'formats|genres', urlKeyHierarchy: '.+' }
     schemes: [https]
 
-programme_clips:
-    path: /programmes/{pid}/clips
-    defaults: { _controller: FrameworkBundle:Redirect:redirect, route: '' }
-    requirements: { pid: '[0-9b-df-hj-np-tv-z]{8,15}' }
-    schemes: [https]
+#programme_clips:
```
Comment (Tasiobg) [inline config/routes/3rd_party.yaml]: I think this is safe to be removed becaseu you added the route inside routes.yaml, or is there any reason to keep it commented out?

Comment (cisco87) [reply config/routes/3rd_party.yaml]: I just forgot it ^^ will remove it

**Comment by BjornTwachtmann** [comment]
There's a fair bit of unused/commented out code in this PR. Did you want us to look at the concept and give feedback, or do you want to tidy it up first?

**Comment by cisco87** [comment]
Please test this

**Comment by cisco87** [comment]
Test this please

**Comment by cisco87** [comment]
test this please

File: templates/clips/list.html.twig [hunk 2]
```diff
@@ -0,0 +1,91 @@
+{% extends 'base_ds2013.html.twig' %}
+
+{% import _self as self %}
```
Comment (BjornTwachtmann) [inline templates/clips/list.html.twig]: Do you need this?

Comment (cisco87) [reply templates/clips/list.html.twig]: I am attached to it (joking i will remove it, was part of another approach to rendering a single clip)

File: src/Controller/Clips/ListController.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+use App\Controller\BaseController;
+use App\Ds2013\Presenters\Utilities\Paginator\PaginatorPresenter;
+use BBC\ProgrammesPagesService\Domain\Entity\Brand;
+use BBC\ProgrammesPagesService\Domain\Entity\Programme;
+use BBC\ProgrammesPagesService\Service\ProgrammesAggregationService;
+use BBC\ProgrammesPagesService\Service\ProgrammesService;
+
+class ListController extends BaseController
+{
+    private const DISPLAYED_ITEMS_LIMIT = 20;
```
Comment (BjornTwachtmann) [inline src/Controller/Clips/ListController.php]: This should be 24

File: src/Controller/Clips/ListController.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    public function __invoke(
+        Programme $programme,
+        ProgrammesAggregationService $aggregationService,
+        ProgrammesService $programmesService
+    ) {
+        $this->setIstatsProgsPageType('programme_clips');
```
Comment (BjornTwachtmann) [inline src/Controller/Clips/ListController.php]: Have you done a comparison of the iStats output of this with the V2 clip page?

File: src/Ds2013/PresenterFactory.php [hunk 5]
```diff
@@ -30,6 +30,7 @@
 use App\Ds2013\Presenters\Pages\EpisodeGuideList\EpisodeGuideListPresenter;
 use App\Ds2013\Presenters\Pages\Schedules\NoSchedule\NoSchedulePresenter;
 use App\Ds2013\Presenters\Section\Clip\Details\ClipDetailsPresenter;
+use App\Ds2013\Presenters\Section\Clip\ListItem\ClipListItemPresenter;
```
Comment (BjornTwachtmann) [inline src/Ds2013/PresenterFactory.php]: Can you revert these changes if you're not using them?

File: templates/clips/list.html.twig [hunk 6]
```diff
@@ -0,0 +1,89 @@
+{% extends 'base_ds2013.html.twig' %}
+
+{% block page_classes %}programmes-page text-base programmes-page--flush{% endblock %}
+{% block title %}{{ meta_context.titlePrefix() }}{% endblock %}
+
+{% block body %}
+    {{ advert('leaderboard', [1,2,3,4], 'programme-advert--leaderboard-wrap') }}
+    <div class="text-base programmes-page programmes-page--smallpush b-g-p" role="main">
```
Comment (BjornTwachtmann) [inline templates/clips/list.html.twig]: `programmes-page prog-layout__primary br-box-page` as per V2

File: templates/clips/list.html.twig [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        <div id="programmes-content" class="skip-to-content-destination" tabindex="-1">Main content</div>
+        <div class="island br-box-secondary">
+            <h1>
+                {{ programme.getTitle() }}
```
Comment (BjornTwachtmann) [inline templates/clips/list.html.twig]: You need to use the entityContextPresenter here to make sure ancestor titles are rendered. The title of the page does not include the title of the current programme here, but it does include the word clips. It is not in a br-box-secondary either, this looks completely different to the live site.

File: templates/clips/list.html.twig [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+                                {% if showAllPid is defined %}
+                                    <a class="box-link" href="{{ path('programme_clips', {pid: showAllPid}) }}">
+                                        {{ tr('All') }}
+                                    </a>
+                                {% else %}
+                                    <span class="box-link">{{ tr('All') }}</span>
+                                {% endif %}
+                            </li>
+
+                            {% for clipsFilter in clipsFilters %}
```
Comment (BjornTwachtmann) [inline templates/clips/list.html.twig]: Would "Series" be more descriptive than "clipsFilters"?

File: templates/clips/list.html.twig [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+                        </ul>
+                    </div>
+                </div>
+            {% endif %}
+        </div>
+        <div class="component component--box--primary" id="parentClips">
+            <div class="component__body br-box-page">
+                <ul class="grid-wrapper highlight-box-wrapper highlight-box-wrapper--grid@bpb1 highlight-box-wrapper--grid@bpw">
+                    {% for clip in clips %}
+                        <li class="grid 1/2@bpb1 1/2@bpw 1/4@bpe">
```
Comment (BjornTwachtmann) [inline templates/clips/list.html.twig]: Have a look at this page in the responsive view. The number of clips in a row at certain breakpoints is completely different to V2

File: templates/clips/list.html.twig [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+                                </li>
+                            {% endfor %}
+                        </ul>
+                    </div>
+                </div>
+            {% endif %}
+        </div>
+        <div class="component component--box--primary" id="parentClips">
+            <div class="component__body br-box-page">
+                <ul class="grid-wrapper highlight-box-wrapper highlight-box-wrapper--grid@bpb1 highlight-box-wrapper--grid@bpw">
```
Comment (BjornTwachtmann) [inline templates/clips/list.html.twig]: Not the same markup as V2.

File: templates/clips/list.html.twig [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+                    {% for clip in clips %}
+                        <li class="grid 1/2@bpb1 1/2@bpw 1/4@bpe">
+                            {# @TODO is this the correct link_location? #}
+                            {{ ds2013('programme', clip, {
+                                'context_programme': programme,
+                                'highlight_box_classes':'highlight-box--list highlight-box--grid@bpb1 programme--grid@bpb1 highlight-box--grid@bpw programme--grid@bpw',
```
Comment (BjornTwachtmann) [inline templates/clips/list.html.twig]: Are these right?

File: templates/clips/list.html.twig [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+                                    'default_width': 240,
+                                    'sizes': { 0: 1/2, 768: 1/4, 1008: '240px'},
+                                },
+                                'title_options': {
+                                    'link_location_track': 'component_clips_carousel',
+                                    'title_format': 'item::ancestry'
+                                },
+                                'body_options': {
+                                    'show_synopsis': false,
+                                    'wordy_duration': false,
```
Comment (BjornTwachtmann) [inline templates/clips/list.html.twig]: You don't need this.

File: templates/clips/list.html.twig [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+                            {{ ds2013('programme', clip, {
+                                'context_programme': programme,
+                                'highlight_box_classes':'highlight-box--list highlight-box--grid@bpb1 programme--grid@bpb1 highlight-box--grid@bpw programme--grid@bpw',
+                                'image_options': {
+                                    'classes': '1/1@bpb1 1/1@bpw',
+                                    'default_width': 240,
+                                    'sizes': { 0: 1/2, 768: 1/4, 1008: '240px'},
+                                },
+                                'title_options': {
+                                    'link_location_track': 'component_clips_carousel',
```
Comment (BjornTwachtmann) [inline templates/clips/list.html.twig]: Where is this link location from? Doesn't seem right here

File: templates/clips/list.html.twig [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+                            </li>
+
+                            {% for clipsFilter in clipsFilters %}
+                                <li>
+                                    {% if clipsFilter.getAvailableClipsCount() > 0 %}
+                                        <a class="box-link" href="{{ path('programme_clips', {pid: clipsFilter.getPid()}) }}">
+                                            {{ clipsFilter.getTitle() }}
+                                        </a>
+                                    {% else %}
+                                        <span>{{ clipsFilter.getTitle }}</span>
```
Comment (BjornTwachtmann) [inline templates/clips/list.html.twig]: getTitle()

File: templates/clips/list.html.twig (hunk 8 above)
Comment (cisco87) [reply templates/clips/list.html.twig]: If you think so I can change it, but they're not always seasons, if you go in a clip it seems to be related to something else (like an episode) so I guess clipsFilters have been the one who can generically describe them in any context?

File: templates/clips/list.html.twig [hunk 15]
```diff
@@ -0,0 +1,97 @@
+{% extends 'base_ds2013.html.twig' %}
+
+{% block page_classes %}text-base programmes-page--smallpush prog-layout{% endblock %}
+{% block title %}{{ meta_context.titlePrefix() }}{% endblock %}
```
Comment (BjornTwachtmann) [inline templates/clips/list.html.twig]: Needs `- tr('clips')` on the end of it
